reference-delegate-explainer.md

```diff
@@ -0,0 +1,534 @@
+
+The Reference Delegate proposal is based heavily on @Westbrook's [Cross-root ARIA Reflection API](https://github.com/Westbrook/cross-root-aria-reflection/blob/main/cross-root-aria-reflection.md) propsal, as well as borrowing ideas from @alice's [Semantic Delegate](https://github.com/alice/aom/blob/gh-pages/semantic-delegate.md) proposal.
+
+A complete solution to cross-root ARIA combines the existing [**ARIAMixin IDL attributes**](https://w3c.github.io/aria/#ARIAMixin) (such as `ariaLabelledbyElements` and `ariaActiveDescendantElement`) to solve the "Referring from Shadow DOM outwards" problem, and introduce a new feature **Reference Delegate** to solve the "Referring into Shadow DOM" problem, which is compatible with the ARIA.
+
+## Proposal
+
+### Reference Delegate
+
+Reference Delegate is a new feature that enables creating ARIA links to elements inside a component's shadow DOM, while maintaining encapsulation of the internal details of the shadow DOM.
+
+**Goals**
+
+- Create a mechanism for ID reference attributes like `aria-activedescendant` and `for` to refer to an element inside a component's shadow DOM.
+- Should be compatible with the ARIAMixin attributes such as `ariaActiveDescendantElement`, to create a complete solution to cross-root ARIA.
+- Should work the same for both closed and open shadow roots.
+- Shadow DOM encapsulation should be preserved: No direct access to any elements inside the shadow tree, and no implementation details leaked into a web component's API.
+- Should allow creating references into multiple nested shadow roots, and across "sibling" shadow roots that don't have a direct parent/child relationship.
+- The solution should be serializable, i.e. support declarative syntax that is expressible in HTML only.
+
+**Non-Goals**
+
+- This is scoped to only solve the problem of referring _into_ the shadow DOM. It relies on ARIAMixin to refer _out_ of the shadow DOM.
+- This proposal does not delegate ARIA attributes that aren't ID references, such as `aria-label`.
+
+### The `referenceDelegate` attribute
+
+A component can specify an element in its shadow tree to act as its "reference delegate". When the host component is the target of a IDREF like a label's `for` attribute, the delegate becomes the true target.
+
+The shadow root specifies the ID of the delegate element inside the shadow DOM. This is done either in JavaScript with the `referenceDelegate` attribute on the `ShadowRoot` object, or in HTML markup using the `shadowrootreferencedelegate` attribute on the `<template>` element.
+
+JavaScript example:
+
+```html
+<template id="t-fancy-input">
+  <input id="real-input" />
+</template>
+
+<script>
+customElements.define("fancy-input", 
+  class FancyInput extends HTMLElement {
+    constructor() {
+      super();
+      this.shadowRoot_ = this.attachShadow({ mode: "closed" });
+      this.shadowRoot_.appendChild(
+        document.getElementById("t-fancy-input").content.cloneNode(true));
```

Comment:
```suggestion
      this.shadowRoot_.innerHTML = '<input id="real-input">'
```

This does the same thing, but without needing the `<template>` above or to clone it.

I think it makes the example a bit clearer, since otherwise someone may be confused by the `real-input` ID elsewhere in the DOM, or the fact that this component is reaching into the global scope.